src/js/jsx/sections/pages/PagesPanel.jsx

```diff
@@ -113,6 +140,27 @@ define(function (require, exports, module) {
             if (!newSelection.isEmpty()) {
                 this._updateLowestNode();
             }
+            
+            if (prevProps.document.id !== this.props.document.id) {
+                // For all layer refs, ask for their registration info and add to list
+                var batchRegistrationInformation = [];
+
+                this.props.document.layers.allVisible.forEach(function (i) {
```

Comment:
This should just be a `map` instead of a `forEach`. If you need it to be an array at the end, call `.toArray`, but even better if you can just send the `Immutable.Iterable` around. (That is, don't convert to an array unless it's actually necessary.)